int8 empty_data
---
int8 success_flag